test/e2e/multicast_test.go

```diff
@@ -0,0 +1,307 @@
+				port:            3460,
+				group:           net.ParseIP("224.3.4.9"),
+			},
+			{
+				name:            "testMulticastMultipleReceiversForInterNodePods",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{1, false}, {2, false}},
+				port:            3461,
+				group:           net.ParseIP("224.3.4.10"),
+			},
+			{
+				name:            "testMulticastMultipleReceiversTrafficFromExternal",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: true},
+				receiverConfigs: []multicastTestPodConfig{{1, false}, {2, true}},
+				port:            3462,
+				group:           net.ParseIP("224.3.4.11"),
+			},
+			{
+				name:            "testMulticastMultipleReceiversTrafficToExternal",
+				senderConfig:    multicastTestPodConfig{nodeIdx: 0, isHostNetwork: false},
+				receiverConfigs: []multicastTestPodConfig{{1, true}, {2, false}},
+				port:            3463,
+				group:           net.ParseIP("224.3.4.12"),
+			},
+		}
+		for _, mc := range testcases {
+			mc := mc
+			t.Run(mc.name, func(t *testing.T) {
+				runTestMulticastBetweenPods(t, data, mc, nodeMulticastInterfaces)
+			})
+		}
+	})
+	t.Run("runTestMulticastForwardToMultipleInterfaces", func(t *testing.T) {
+		multipleInterfacesFound := false
+		var nodeIdx int
+		for i, ifaces := range nodeMulticastInterfaces {
+			if len(ifaces) >= 2 {
+				multipleInterfacesFound = true
+				nodeIdx = i
+				break
+			}
+		}
+		if !multipleInterfacesFound {
+			t.Skip("Skipping test because none of the Nodes has more than one multicast enabled interface")
+		}
+		runTestMulticastForwardToMultipleInterfaces(t, data, nodeIdx, 3464, "224.3.4.13", nodeMulticastInterfaces[nodeIdx])
+	})
+}
+
+type multicastTestPodConfig struct {
+	nodeIdx       int
+	isHostNetwork bool
+}
+
+type multicastTestcase struct {
+	name            string
+	senderConfig    multicastTestPodConfig
+	receiverConfigs []multicastTestPodConfig
+	port            int
+	group           net.IP
+}
+
+func runTestMulticastForwardToMultipleInterfaces(t *testing.T, data *TestData, senderIdx int, senderPort int, senderGroup string, senderMulticastInterfaces []string) {
+	senderName, _, cleanupFunc := createAndWaitForPod(t, data, data.createMcJoinPodOnNode, "test-sender-", nodeName(senderIdx), testNamespace, false)
+	defer cleanupFunc()
+	tcpdumpName, _, cleanupFunc := createAndWaitForPod(t, data, data.createNetshootPodOnNode, "test-tcpdump-", nodeName(senderIdx), testNamespace, true)
+	defer cleanupFunc()
+	// Wait 2 seconds(-w 2) before sending multicast traffic.
+	// It sends two multicast packets for every second(-f 500 means it takes 500 milliseconds for sending one packet).
+	sendMulticastCommand := []string{"/bin/sh", "-c", fmt.Sprintf("mcjoin -f 500 -o -p %d -s -t 3 -w 2 %s", senderPort, senderGroup)}
+	go func() {
+		data.runCommandFromPod(testNamespace, senderName, mcjoinContainerName, sendMulticastCommand)
```

Comment:
How this goroutine terminates?

Reply:
I think it will terminate when mcjoin pod gets deleted. There doesn't not like a good practice for me so I added -W timeout here